internal/controllers/tesla.go

```diff
@@ -431,90 +432,80 @@ func (tc *TeslaController) GetVirtualKeyStatus(c *fiber.Ctx) error {
 		return fiber.NewError(fiber.StatusInternalServerError, "Error checking fleet status.")
 	}
 
-	fleetTelemetryCapable := IsFleetTelemetryCapable(fleetStatus)
+	fleetTelemetryCapable := service.IsFleetTelemetryCapable(fleetStatus)
 
 	var response = VirtualKeyStatusResponse{}
 	response.Added = fleetStatus.KeyPaired
 	if !fleetTelemetryCapable {
-		response.Status = Incapable
+		response.Status = service.Incapable
 	} else if fleetStatus.KeyPaired {
-		response.Status = Paired
+		response.Status = service.Paired
 	} else {
-		response.Status = Unpaired
+		response.Status = service.Unpaired
 	}
 
 	return c.JSON(response)
 }
 
-type VirtualKeyStatusResponse struct {
-	Added  bool             `json:"added"`
-	Status VirtualKeyStatus `json:"status" swaggertype:"string"`
-}
-
-type VirtualKeyStatus int
-
-const (
-	Incapable VirtualKeyStatus = iota
-	Paired
-	Unpaired
-)
+// GetFleetStatus godoc
+// @Summary     Get fleet status
+// @Description Retrieves detailed fleet status information for a Tesla vehicle, including virtual key connection status and telemetry capabilities.
+// @Tags        tesla,fleet
+// @Accept      json
+// @Produce     json
+// @Param       vin query string true "Vehicle VIN"
+// @Security    BearerAuth
+// @Success     200 {object} service.VehicleFleetStatus "Fleet status details"
+// @Failure     400 {object} fiber.Error "Bad Request"
+// @Failure     401 {object} fiber.Error "Unauthorized"
+// @Failure     500 {object} fiber.Error "Internal server error"
+// @Router      /v1/tesla/fleet-status [get]
+func (tc *TeslaController) GetFleetStatus(c *fiber.Ctx) error {
+	walletAddress := helpers.GetWallet(c)
 
-func (s VirtualKeyStatus) String() string {
-	switch s {
-	case Incapable:
-		return "Incapable"
-	case Paired:
-		return "Paired"
-	case Unpaired:
-		return "Unpaired"
+	var params VinInput
+	if err := c.QueryParser(&params); err != nil {
+		tc.logger.Err(err).Msgf("Failed to parse request URL params")
+		return fiber.NewError(fiber.StatusBadRequest, "Failed to parse request URL params")
 	}
-	return ""
-}
-
-func (s VirtualKeyStatus) MarshalText() ([]byte, error) {
-	return []byte(s.String()), nil
-}
 
-func (s *VirtualKeyStatus) UnmarshalText(text []byte) error {
-	switch str := string(text); str {
-	case "Incapable":
-		*s = Incapable
-	case "Paired":
-		*s = Paired
-	case "Unpaired":
-		*s = Unpaired
-	default:
-		return fmt.Errorf("unrecognized status %q", str)
+	sd, err := tc.teslaService.GetVehicleByVIN(c.Context(), tc.logger, tc.pdb, params.VIN)
+	if err != nil {
+		return fiber.NewError(fiber.StatusInternalServerError, "Failed to get vehicle by VIN.")
```

Comment:
Is it an internal error if the VIN is not found?

Reply:
good catch